hpx/runtime/serialization/vector.hpp

```diff
@@ -23,13 +23,11 @@ namespace hpx { namespace serialization
         ar >> size; //-V128
         if(size == 0) return;
 
-        vs.reserve(size);
+        vs.resize(size);
```

Comment:
You can see that for the old version there is already default construction of temporary object inside the loop ```value_type v;``` , which later is moved to vector by push_back. This commit is only about avoiding this extra moving by default construction of objects and their deserialization right inside the vector.